src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -2671,20 +3162,20 @@ void M3()
         S_WithSemiAutoProperty s2 = s1;
     }
 }
-");
+", parseOptions: TestOptions.RegularNext.WithFeature("run-nullable-analysis", runNullableAnalysis));
             var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
             comp.TestOnlyCompilationData = accessorBindingData;
             Assert.Equal("System.Int32 S_WithAutoProperty.<P>k__BackingField", comp.GetTypeByMetadataName("S_WithAutoProperty").GetMembers().OfType<FieldSymbol>().Single().ToTestDisplayString());
             Assert.Empty(comp.GetTypeByMetadataName("S_WithManualProperty").GetMembers().OfType<FieldSymbol>());
             Assert.Empty(comp.GetTypeByMetadataName("S_WithSemiAutoProperty").GetMembers().OfType<FieldSymbol>());
             comp.VerifyDiagnostics(
-                // PROTOTYPE(semi-auto-props): Do we expect a similar error for semi auto props?
                 // (22,33): error CS0165: Use of unassigned local variable 's1'
                 //         S_WithAutoProperty s2 = s1;
-                Diagnostic(ErrorCode.ERR_UseDefViolation, "s1").WithArguments("s1").WithLocation(22, 33)
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "s1").WithArguments("s1").WithLocation(22, 33),
+                // (34,37): error CS0165: Use of unassigned local variable 's1'
+                //         S_WithSemiAutoProperty s2 = s1;
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "s1").WithArguments("s1").WithLocation(34, 37)
             );
-
-            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
```

Comment:
I'm not able to properly assert `NumberOfPerformedAccessorBinding`.

To my understanding, sometimes `MethodCompiler.CompileMethod` is called with `M3` before the semi property accessor, and sometimes the opposite. This affects whether a force binding was performed.

I'm not sure about the root cause of this, but in `CompileNamespace`, I see the following:

```csharp
        private void CompileNamespace(NamespaceSymbol symbol)
        {
            foreach (var s in symbol.GetMembersUnordered())
            {
                s.Accept(this, null);
            }
        }
```

If I replace `GetMembersUnordered` with `GetMembers`, things get more stable.

Reply:
That is likely to explain the behavior. Consider putting type C into its own syntax tree and requesting diagnostics for each tree separately. That should make the result deterministic. We probably should test both orders and observe difference in behavior between them.

Reply:
Nice! That worked well locally. Will see what happens in CI.